pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -77,12 +78,18 @@
   private static final Long DATA_DIRECTORY_EXCEPTION_VALUE = 1100000L;
   private static final String METADATA_EVENT_NOTIFIER_PREFIX = "metadata.event.notifier";
 
+  // Note: Right now we don't allow pinot-only mode to be used in production yet.
+  // We'll have this mode used only in tests.
+  // TODO: Once all the logic is in then we can remove this flag.
+  protected boolean _enablePinotOnlyMode = false;
```

Comment:
I like introducing a protected method (like: boolean isPinotOnlyModeSupported()) and overriding it in tests.

Reply:
Done.